### PR TITLE
[Feature] : Add grid_sampler TensorRT Plugin

### PR DESCRIPTION
**Pull request by grimoire**
This PR add grid_sampler TensorRT plugin.
Please register symbolics before converting onnx.
```
from mmcv.onnx.symbolic import register_extra_symbolics
register_extra_symbolics(11)
```

### CODE REVIEW

**Comment by Johnson-Wang** [comment]
Hi, thanks for your contribution. I wonder if it is okay to create a relevant PR in mmdetection for mask-rcnn conversion.

File: tests/test_ops/test_tensorrt.py [hunk 1]
```diff
@@ -326,3 +326,70 @@ def func(data):
...
     assert torch.allclose(pytorch_results, trt_results)
+
+
+def test_grid_sample():
+    from mmcv.onnx.symbolic import register_extra_symbolics
+
+    register_extra_symbolics(11)
+
+    def func(input, grid):
+        return nn.functional.grid_sample(input, grid)
```
Comment (Johnson-Wang) [inline tests/test_ops/test_tensorrt.py:337]: Other interpolation and `align_corners` modes should also be tested.

**Comment by grimoire** [comment]
@Johnson-Wang Sure! That would be cool!

File: tests/test_ops/test_tensorrt.py [hunk 2]
```diff
@@ -402,3 +402,88 @@ def test_deform_conv():
     if os.path.exists(trt_file):
         os.remove(trt_file)
     assert torch.allclose(pytorch_results, trt_results)
+
+
```
Comment (Johnson-Wang) [inline tests/test_ops/test_tensorrt.py:406]: Add the following decorator instead of creating a `params_list` would be clearer. 
```python
@pytest.mark.parametrize('mode', ['bilinear', 'nearest'])
@pytest.mark.parametrize('padding_mode', ['zeros', 'border', 'reflection'])
@pytest.mark.parametrize('align_corners', [True, False])
def test_grid_sample(mode, padding_mode, align_corners):
```

**Review by Johnson-Wang — APPROVED** [review]
The fp16 implement is not emergent for now. Therefore, this PR is thought ready to merge.

File: mmcv/ops/csrc/tensorrt/plugins/trt_grid_sampler_kernel.cu [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+#include "common_cuda_helper.hpp"
+#include "trt_cuda_helper.cuh"
+#include "trt_grid_sampler.hpp"
+#include "trt_plugin_helper.hpp"
+
+using mmcv::GridSamplerInterpolation;
+using mmcv::GridSamplerPadding;
+using mmcv::TensorDesc;
+
+//// the code copy from
```
Comment (ZwwWayne) [inline mmcv/ops/csrc/tensorrt/plugins/trt_grid_sampler_kernel.cu:17]: Should this be put to the top of the file?

File: mmcv/ops/csrc/tensorrt/plugins/trt_grid_sampler.cpp [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+#include <stdio.h>
+
+#include <chrono>
+
+#include "trt_serialize.hpp"
+
+using mmcv::GridSamplerInterpolation;
+using mmcv::GridSamplerPadding;
+
+void grid_sample_float(float *output, const float *input, const float *grid,
```
[the remark this answers is not in the document]
Comment (ZwwWayne) [reply mmcv/ops/csrc/tensorrt/plugins/trt_grid_sampler.cpp:13]: ping this comment

**Review by Johnson-Wang — APPROVED** [review]

File: mmcv/ops/csrc/tensorrt/plugins/trt_grid_sampler.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (Johnson-Wang) [reply mmcv/ops/csrc/tensorrt/plugins/trt_grid_sampler.cpp:13]: The fp16 implement is not emergent for now since the `grid_sample` op is forced to be fp32 in mmdet.
